core/tasks/channels/cron.go

```diff
@@ -0,0 +1,94 @@
+package channels
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/nyaruka/gocommon/dates"
+	"github.com/nyaruka/mailroom/core/models"
+	"github.com/nyaruka/mailroom/core/msgio"
+	"github.com/nyaruka/mailroom/core/tasks"
+	"github.com/nyaruka/mailroom/runtime"
+)
+
+func init() {
+	tasks.RegisterCron("sync_old_seen_android_channels", &SyncOldSeenAndroidCron{})
+}
+
+type SyncOldSeenAndroidCron struct {
+	FCMClient msgio.FCMClient
+}
+
+func (s *SyncOldSeenAndroidCron) AllInstances() bool {
+	return true
+}
+
+func (s *SyncOldSeenAndroidCron) Next(last time.Time) time.Time {
+	return tasks.CronNext(last, time.Minute*5)
+
+}
+
+func (s *SyncOldSeenAndroidCron) Run(ctx context.Context, rt *runtime.Runtime) (map[string]any, error) {
+
+	if s.FCMClient == nil {
+		s.FCMClient = msgio.CreateFCMClient(ctx, rt.Config)
+	}
+
+	oldSeenAndroidChannels, err := getOldSeenAndroidChannels(ctx, rt.DB)
+	if err != nil {
+		return nil, fmt.Errorf("error loading old seen android channels: %w", err)
+	}
+
+	erroredCount := 0
+	syncedCount := 0
+
+	for _, channel := range oldSeenAndroidChannels {
+		err := msgio.SyncAndroidChannel(ctx, s.FCMClient, &channel)
+		if err != nil {
+			slog.Error("error syncing messages", "error", err, "channel_uuid", channel.UUID())
+			erroredCount += 1
+		} else {
+			syncedCount += 1
+		}
+
+	}
+
+	return map[string]any{"synced": syncedCount, "errored": erroredCount}, nil
+
+}
+
+func getOldSeenAndroidChannels(ctx context.Context, db models.DBorTx) ([]models.Channel, error) {
```

Comment:
we could easily move this to models but currently that only needed for this cron only

Reply:
We should move it to models.. it's not just a question of re-use but grouping code that does similar stuff. If we make changes to the db schema of channel for example - it's easier if the functions that query channels from the database are in the same place and their sql are in the same file.